frontend/ui/widget/dictquicklookup.lua

```diff
@@ -794,12 +796,60 @@ function DictQuickLookup:update()
     end
 
     -- Update main text widgets
-    if self.is_html then
+    if self.is_html and self.shw_widget then
         self.text_widget.htmlbox_widget:setContent(self.definition, self:getHtmlDictionaryCss(), Screen:scaleBySize(self.dict_font_size))
-    else
+        -- Scroll back to top
+        self.text_widget.htmlbox_widget.page_number = 1
+        self.text_widget.v_scroll_bar:set((self.text_widget.htmlbox_widget.page_number-1) / self.text_widget.htmlbox_widget.page_count, self.text_widget.htmlbox_widget.page_number / self.text_widget.htmlbox_widget.page_count)
+        self.text_widget.v_scroll_bar.enable = self.text_widget.htmlbox_widget.page_count > 1
+    elseif not self.is_html and self.stw_widget then
         self.text_widget.text_widget.text = self.definition
         -- NOTE: The recursive free via our WidgetContainer (self[1]) above already free'd us ;)
         self.text_widget.text_widget:init()
+        -- Scroll back to top
+        local low, high = self.text_widget.text_widget:getVisibleHeightRatios()
+        self.text_widget.v_scroll_bar:set(low, high)
+        local visible_line_count = self.text_widget:getVisLineCount()
+        local total_line_count = self.text_widget:getAllLineCount()
+        self.text_widget.v_scroll_bar.enable = visible_line_count < total_line_count
+    else
+        -- We jumped from HTML to Text of vice-versa, we need a new widget instance
+        -- Whee, code duplication! (this is copied verbatim from init)
+        if self.is_html then
+            self.shw_widget = ScrollHtmlWidget:new{
```

Comment:
I don't like this code duplication, for having to tweaks the params in both places and we might easily forget the other, and having to duplicate comments, etc...
Can't we create both Scroll*Widget empty (or ready enough) in init, and re-use/swap them as needed ?

Reply:
That would probably work. I tried without doing that because, arguably, those are "heavy" widgets, but it's probably moot in the grand scheme of things ;).

Reply:
Thought we could even push that further and have these Scroll*Widget instances stored not in `self`, but in the single DictQuickLookup class object, so the 2 initial instances would just be re-used by all further lookups - and may be this would fix the memory usage/leak issue I suspect happens with dicts.

But... we can have stacked DictQuickLookups when re-looking up inside a dict :( So, no go.